### PR TITLE
Unwrap `AssertionError` from `InvocationTargetException` during Quarkus JUnit5 callbacks

### PR DESCRIPTION
**Pull request by msavy**
I have been writing some custom Quarkus JUnit 5 callbacks; these are specially provided by Quarkus to allow callbacks within the context of the Quarkus classloader to do things like CDI, etc.

I've generally managed to get most things working after experimentation, plus digging through code and documentation, but there was one nit:

Currently, only `RuntimeException`-derived throwables are being unwrapped, but many test frameworks use throwables deriving from `AssertionError`. Hence, if an `AssertionError` is thrown in the callback, it currently ends up being wrapped by an `InvocationTargetException`, which means that tooling doesn't parse the stack trace properly for "expected" vs "actual" comparisons, etcetera.

This small pull request ensures that `AssertionError` is unwrapped properly (but not `Error` more generically, as this could mask categories of error that emanate from other sources such as OOM, where leaving it unwrapped may be helpful).

I added a test that exercises the existing codepaths as well as the new one.

p.s: I initially tried to add the tests to existing `@QuarkusTest`s that cover callbacks, but despite sinking many hours into it, I couldn't find a sensible way to capture the exceptions thrown from the Quarkus callbacks inside of the JUnit test, so this semi-mocked approach seems the most practical way without significant code changes, which I doubt are worth it.

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by msavy** [comment]
@geoand CI doesn't seem to be triggering for some reason

**Comment by msavy** [comment]
GH workflows says it is still waiting for approval from a maintainer, which you are, so not sure why it's not picking it up.

**Comment by geoand** [comment]
Done!

**Pull request merged**
